sor/src/main/java/com/bazaarvoice/emodb/sor/db/astyanax/AstyanaxDataReaderDAO.java

```diff
@@ -573,15 +580,68 @@ private void addScanRangeSplitsForTokenRange(CassandraKeyspace keyspace, ColumnF
         // Split the token range into sub-ranges with approximately the desired number of records per split
         String rangeStart = tokenRange.getStartToken();
 
-        List<CfSplit> splits = getCfSplits(
-                keyspace.getAstyanaxKeyspace(), cf, tokenRange.getStartToken(), tokenRange.getEndToken(),
-                desiredRecordsPerSplit, allTokenRanges);
+        Deque<ScanRangeSplitWorkItem> splitWorkQueue = new LinkedList<>();
+        splitWorkQueue.push(new ScanRangeSplitWorkItem(tokenRange, desiredRecordsPerSplit, true));
+
+        ScanRangeSplitWorkItem splitWork;
+        while ((splitWork = splitWorkQueue.poll()) != null) {
+            try {
+                List<CfSplit> splits = getCfSplits(
+                        keyspace.getAstyanaxKeyspace(), cf, splitWork.range.getStartToken(), splitWork.range.getEndToken(),
+                        splitWork.desiredRecordsPerSplit, allTokenRanges);
+
+                for (CfSplit split : splits) {
+                    if (splitWork.desiredRecordsPerSplit <= desiredRecordsPerSplit) {
+                        ByteBuffer begin = parseTokenString(split.getStartToken());
+                        ByteBuffer finish = parseTokenString(split.getEndToken());
+                        builder.addScanRange(rack, rangeStart, ScanRange.create(begin, finish));
+                    } else {
+                        // This work item was for a larger-than-desired split created due to a previous timeout.
+                        // Add the split back to the work queue to be split again at a smaller size.  Note that
+                        // retryOnTimeout is set to false since we've already established that growing and subdividing
+                        // it won't help.
+                        TokenRange newWorkTokenRange = new TokenRangeImpl(split.getStartToken(), split.getEndToken(), splitWork.range.getEndpoints());
+                        int newWorkDesiredSize = splitWork.desiredRecordsPerSplit / 10;
+                        splitWorkQueue.push(
+                                new ScanRangeSplitWorkItem(newWorkTokenRange, newWorkDesiredSize, false));
+                        _log.debug("Decreasing scan range split to {} for keyspace {} and range {}", newWorkDesiredSize, keyspace.getName(), newWorkTokenRange);
+                    }
+                }
+            } catch (Exception e) {
+                if (isTimeoutException(e)) {
+                    if (splitWork.retryOnTimeout) {
+                        // Try again with 10 times the desired number of records per split, up to a reasonable maximum
+                        int retryDesiredRecordsPerSplit = (int) Math.min(splitWork.desiredRecordsPerSplit * 10L, Integer.MAX_VALUE);
```

Comment:
We could monitor the logs in case of the 1st time split failure days, and see if 10 is appropriate or any other value is better. so, may be a  static variable for the value 10 to be used in this line and line:603?